packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/ModuleHolder.kt

```diff
@@ -48,18 +48,21 @@ class ModuleHolder<T : Module>(val module: T) {
       // Give the module object a name. It's used for compatibility reasons, see `EventEmitter.ts`.
       moduleDecorator.registerProperty("__expo_module_name__", false, emptyArray(), { name }, false, emptyArray(), null)
 
-      val viewFunctions = definition.viewManagerDefinition?.asyncFunctions
-      if (viewFunctions?.isNotEmpty() == true) {
-        trace("Attaching view prototype") {
-          val viewDecorator = JSDecoratorsBridgingObject(jniDeallocator)
-          viewFunctions.forEach { function ->
-            function.attachToJSObject(appContext, viewDecorator, "${name}_${definition.viewManagerDefinition?.viewType?.name}")
+      definition.viewManagerDefinitions.values.forEach {
+        val viewFunctions = it.asyncFunctions
+        if (viewFunctions.isNotEmpty() == true) {
+          trace("Attaching view prototype") {
+            val viewDecorator = JSDecoratorsBridgingObject(jniDeallocator)
+            viewFunctions.forEach { function ->
+              function.attachToJSObject(appContext, viewDecorator, "${name}_${it.name}")
+            }
+
+            moduleDecorator.registerObject("ViewPrototype", viewDecorator)
           }
```

Comment:
```suggestion
      definition.viewManagerDefinitions.values.forEach { definition ->
        val viewFunctions = definition.asyncFunctions
        if (viewFunctions.isNotEmpty() == true) {
          trace("Attaching view prototype") {
            val viewDecorator = JSDecoratorsBridgingObject(jniDeallocator)
            viewFunctions.forEach { function ->
              function.attachToJSObject(appContext, viewDecorator, "${name}_${definition.name}")
            }

            moduleDecorator.registerObject("ViewPrototype", viewDecorator)
          }
```
I would also invert if and return if viewFunctions is empty.